markdown_posts/markdown_converter/Converter.java

```diff
@@ -0,0 +1,106 @@
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+
+public class Converter {
+
+  public static void removeBeginning (int index, String[] split, char drop) {
+    int length = split[index].length();
+//remove beginning paren
+    if (split[index] != null && length > 0 && split[index].charAt(0) == drop) {
+      split[index] = split[index].substring(1, length);
+    }
+
+  }
+
+  public static void removeEnding (int index, String[] split, char drop) {
+    int length = split[index].length();
+    if (split[index] != null && length > 0 &&
+          split[index].charAt(length-1) == drop) {
+      split[index] = split[index].substring(0, length-1);
+    }
+
+  }
+
+  public static String replaceSubstring (String toUpdate, CharSequence remove, CharSequence insert) {
+    String replaced = toUpdate.replace(remove, insert);
+
+    return replaced;
+  }
+
+  public static void main(String[] args) throws IOException {
+    String read = null;
+    String[] split = new String[4]; // sorry for the magic number
+    try {
+      BufferedReader rawBlogContents = new BufferedReader(
+                                        new FileReader(args[0]));
+
+      while ((read = rawBlogContents.readLine()) != null) {
+        split = read.split(",'");
+
+        for (int i = 0; i < split.length; ++i) {
+
+//remove beginning paren
+          removeBeginning (i, split, '(');
+//remove ending semicolon
+          removeEnding (i, split, ';');
+//remove ending paren
+          removeEnding (i, split, ')');
+//remove trailing apostrophes
+          removeEnding (i, split, '\'');
+
+//update mistranslated characters
+          split[i] = replaceSubstring (split[i], "&#8217;", "'");
+          split[i] = replaceSubstring (split[i], "â€“", "-");
+          split[i] = replaceSubstring (split[i], "â€™", "'");
+          split[i] = replaceSubstring (split[i], "\\\"", "\"");
+          split[i] = replaceSubstring (split[i], "&#8220;", "\"");
+          split[i] = replaceSubstring (split[i], "&#8221;", "\"");
+          split[i] = replaceSubstring (split[i], "\\\'", "'");
+          //this one will convert browser newlines to markdown newlines
+          split[i] = replaceSubstring (split[i], "\\r\\n", "<br>");
```

Comment:
I don't think we actually want this line. This is converting line breaks to HTML, but the source is already HTML, so it's inserting new line breaks where they don't belong.

Reply:
I didn't notice any extra line breaks -- the \r\n single breaks are being replaced with a single <br>. Did you have a different experience when you ran the code? 

The reason I'm switching to <br> from the return + newline is due to documentation I read around use of HTML newline within markdown files. If the use of <br> does not work then I am planning to change to using markdown line breaks instead (two spaces at the end of a line) in place of the \r\n.

Example of use of <br> in markdown is here: https://stackoverflow.com/questions/13730982/force-line-break-br-in-header-h1-in-markdown